fleetdm/fleet
# feat: add software title ID to add software activity

> Related issue: #24120

## Changes
- Added the `software_title_id` field to the activity details for `added_software` activities, which get generated when adding a customer installer or a FMA
- Added a return value (`titleID`) to `ds.MatchOrCreateSoftwareInstaller`
- Removed `ds.GetSoftwareTitleIDByMaintainedAppID`. Since we're returning the new value above, this method was no longer needed.

## Testing steps
1. Add a custom installer
2. Add a FMA
3. Check the activity details in the response to `GET /activities`. Verify that the `software_title_id` field exists and is correct.
4. Add a FMA with automatic install. Make sure the policy is correctly created, has the correct software title ID associated with it, and that it installs the app.

# Checklist for submitter

If some of the following don't apply, delete the relevant line.



  See [Changes files](https://github.com/fleetdm/fleet/blob/main/docs/Contributing/Committing-Changes.md#changes-files) for more information.
